src/scaffoldmaker/meshtypes/meshtype_3d_lung1.py

```diff
@@ -780,3 +684,496 @@ def refineMesh(cls, meshrefinement, options):
+                    coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_VALUE, 1, x)
+                    coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_D_DS1, 1, d1)
+                    coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_D_DS2, 1, d2)
+                    coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_D_DS3, 1, d3)
+                    lowerNodeIds[n3][n2][n1] = nodeIdentifier
+                    nodeIdentifier += 1
+
+    # Upper lobe nodes
+    for n3 in range(uElementsCount3 + 1):
+        upperNodeIds.append([])
+        for n2 in range(uElementsCount2 + 1):
+            upperNodeIds[n3].append([])
+            for n1 in range(uElementsCount1 + 1):
+                upperNodeIds[n3][n2].append(None)
+                if ((n1 == 0) or (n1 == uElementsCount1)) and ((n2 == 0) or (n2 == uElementsCount2)):
+                    continue
+                if (n2 < (uElementsCount2 - 2)) and (n3 < (uElementsCount3 - 2)):
+                    continue
+                if ((n2 == 0) or (n2 == uElementsCount2)) and (n3 == uElementsCount3):
+                    continue
+                if ((n1 == 0) or (n1 == uElementsCount1)) and (n3 == uElementsCount3):
+                    continue
+
+                if lungSide != diaphramgaticLung:
+                    # Oblique fissure nodes
+                    if (n2 == (uElementsCount2 - 2)) and (n3 < (uElementsCount3 - 2)):
+                        upperNodeIds[n3][n2][n1] = lowerNodeIds[n3][lElementsCount2][n1]
+                        continue
+                    elif (n2 < (uElementsCount2 - 1)) and (n3 == (uElementsCount3 - 2)):
+                        upperNodeIds[n3][n2][n1] = lowerNodeIds[lElementsCount3][n2][n1]
+                        continue
+
+                node = nodes.createNode(nodeIdentifier, nodetemplate)
+                cache.setNode(node)
+                if generateParameters:
+                    x = [1.0 * (n1 - 1) + xMirror, 1.0 * (n2 - 1) + 2.5, 1.0 * n3 + 2.0]
+                else:
+                    nodeParameters = nodeFieldParameters[nodeIndex]
+                    nodeIndex += 1
+                    assert nodeIdentifier == nodeParameters[0]
+                    x, d1, d2, d3 = nodeParameters[1]
+                coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_VALUE, 1, x)
+                coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_D_DS1, 1, d1)
+                coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_D_DS2, 1, d2)
+                coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_D_DS3, 1, d3)
+                upperNodeIds[n3][n2][n1] = nodeIdentifier
+                nodeIdentifier += 1
+
+    return nodeIndex, nodeIdentifier
+
+def getLungElements(lungSide, coordinates, eftfactory, eftRegular, elementtemplateRegular, elementtemplateCustom, mesh,
+                    lungMeshGroup, lungSideMeshGroup, lowerLungMeshGroup, upperLungMeshGroup, middleRightLungMeshGroup,
```

Comment:
I suggest you have just lowerLobeMeshGroup, middleLobeMeshGroup, upperLobeMeshGroup (in that order) and pass None to any of them if you don't want them.
This (and removing special handling for middle) allows pig left lung with 3 lobes and mouse left lung with 1 lobe (i.e. no special groups) to be handled by the same code. And it's more readible.